base/math.jl

```diff
@@ -1245,6 +1245,8 @@ end
 function ^(x::Union{Float16,Float32}, n::Integer)
     n == -2 && return (i=inv(x); i*i)
     n == 3 && return x*x*x #keep compatibility with literal_pow
+    # It won't work to do `inv(x)^-n` if `n` is typemin.
+    n == typemin(typeof(n)) && return (x^cld(n,2))*(x^fld(n,2))
```

Comment:
This stack overflows for unsigned `n` and throws for integers without a `typemin`. Perhaps a better test would be `n < 0 && -n < 0`. As for the computation, I imagine that a power-by-squaring approach would be more efficient: 

```julia
y = (x*x)^fld(n, 2)
isodd(n) ? y * x : y
```

I'd also like to see these two test cases included in the unit tests for this PR (they illustrate the problems I mentioned above):

```julia
@test Float32(1.1)^unsigned(0) === Float32(1.0)
@test Float32(1.1)^big(0) === Float32(1.0)
```

Reply:
Good catch for the integers without typemin.   Tests added.